src/ORM/DataList.php

```diff
@@ -658,6 +659,39 @@ public function filterByCallback(callable $callback): SS_List
         return $output;
     }
 
+    /**
+     * Return a copy of this list which only includes items where $fieldToFilterBy matches values in $fieldInOtherList from $list.
+     *
+     * If both fields are ID, the $list dataclass must be in the same class hierarchy as the dataclass in this list
+     *
+     * @param SS_List<DataObject> $list
+     * @param string $fieldToFilterBy The name of the field in $this to filter by
+     * @param string $fieldInOtherList The name of the field in $list to get filter values from
+     * @return static<T>
+     * @throws InvalidArgumentException
+     */
+    public function filterByList(SS_List $list, string $fieldToFilterBy = 'ID', string $fieldInOtherList = 'ID'): static
+    {
+        $thisDataClass = $this->dataClass();
+        $listDataClass = $this->getDataClassFromList($list);
+        if ($fieldInOtherList === 'ID'
+            && $fieldInOtherList === 'ID'
+            && !is_a($thisDataClass, $listDataClass, true)
+            && !is_a($listDataClass, $thisDataClass, true)
+        ) {
+            throw new InvalidArgumentException('If both columns are ID, the $list dataclass must be in the same class hierarchy as the dataclass in this list');
+        }
+        // Try to filter by the list with a subquery for best performance
+        if ($list instanceof DataList) {
```

Comment:
```suggestion
        if (is_a($list, DataList::class)) {
```
particularly since you just used is_a() above

Reply:
I used `is_a` above because I was checking against a string. Here I'm checking against an object.
This is subjective, and as I say every time you want me to change this, I'll keep my preferred way here.

We don't and won't have a specific requirement to always use `is_a()` instead of `instanceof`.